packages/MJServer/src/generic/ResolverBase.ts

```diff
@@ -548,9 +548,14 @@ export class ResolverBase {
   }
 
   protected ListenForEntityMessages(entityObject: BaseEntity, pubSub: PubSubEngine, userPayload: UserPayload) {
-    if (!this._eventSubscription) {
+    // The unique key is set up for each entity object via it's primary key to ensure that we only have one listener at most for each unique
+    // entity/record in the system. This is important because we don't want to have multiple listeners for the same entity/record as it could
+    // cause issues with multiple messages for the same event.
+    const uniqueKey = `${entityObject.EntityInfo.Name}_${entityObject.PrimaryKey.ToURLSegment()}`;
+
+    if (!ResolverBase._eventSubscriptions.has(entityObject.EntityInfo.Name)) {
       // listen for events from the entityObject in case it is a long running task and we can push messages back to the client via pubSub
-      this._eventSubscription = MJGlobal.Instance.GetEventListener(false).subscribe(async (event) => {
+      const listener = MJGlobal.Instance.GetEventListener(false).subscribe(async (event) => {
```

Comment:
This is more accurately described as a `subscription`– the listener (Observable) is what's returned by `GetEventListener`, whereas what's returned by `subscribe` is a reference to that subscription (so it can later be unsubscribed, usually)